synapseutils/copy.py

```diff
@@ -206,24 +250,23 @@ def _copyFile(syn, entity, destinationId, version=None, updateExisting=False, se
     bundle = syn._getEntityBundle(ent.id, version=ent.versionNumber, bitFlags=0x800|0x1)
     fileHandle = synapseclient.utils.find_data_file_handle(bundle)
     createdBy = fileHandle['createdBy']
-    #CHECK: If the user created the file, copy the file by using fileHandleId else hard copy
+    #CHECK: If the user created the file, copy the file by using fileHandleId else copy the fileHandle
+    path=ent.path
+    store=True
     if profile.ownerId == createdBy:
-        new_ent = File(name=ent.name, parentId=destinationId)
-        new_ent.dataFileHandleId = ent.dataFileHandleId
+        newdataFileHandleId = ent.dataFileHandleId
     else:
+        copiedFileHandle = copyFileHandleId(syn, [fileHandle], ["FileEntity"], [bundle['entity']['id']])
+
+        newdataFileHandleId = copiedFileHandle['copyResults'][0]['newFileHandle']['id']
```

Comment:
Raise error if filehandle didn't copy.  I.e. don't trhown an key not found error when copy failed.